packages/core/src/api/exporters/html/htmlConversion.test.ts

```diff
@@ -101,3 +102,204 @@ describe("Test HTML conversion", () => {
     });
   }
 });
+
+// Fragments created from ProseMirror selections don't always conform to the
+// schema. This is because ProseMirror preserves the full ancestry of selected
+// nodes, but not the siblings of ancestor nodes. These tests are to verify that
+// Fragments like this are exported to HTML properly, as they can't be created
+// from Block objects like all the other test cases (Block object conversions
+// always conform to the schema).
+describe("Test ProseMirror fragment edge case conversion", () => {
+  let editor: BlockNoteEditor;
+  const div = document.createElement("div");
+  beforeEach(() => {
+    editor = BlockNoteEditor.create();
+    editor.mount(div);
+  });
+
+  afterEach(() => {
+    editor.mount(undefined);
+    editor._tiptapEditor.destroy();
+    editor = undefined as any;
+
+    delete (window as Window & { __TEST_OPTIONS?: any }).__TEST_OPTIONS;
+  });
+
+  // When the selection starts in a nested block, the Fragment from it omits the
+  // `blockContent` node of the parent `blockContainer` if it's not also
+  // included in the selection. In the schema, `blockContainer` nodes should
+  // contain a single `blockContent` node, so this edge case needs to be tested.
+  describe("No block content", () => {
+    it("Selection within a block's children", () => {
+      // Fragment created when the selection starts and ends within a block's
+      // children.
+      const copiedFragment = Fragment.from(
```

Comment:
might be a bit cleaner to first create the blocknote json (more readable), convert that to prosemirror (we have a function for this), and then select the relevant part from it?

(I might be wrong here, wdyt?)